granulate_utils/containers/cri.py

```diff
@@ -73,49 +74,56 @@ def list_containers(self, all_info: bool) -> List[Container]:
 
         for rt, path in self._runtimes.items():
             with RuntimeServiceWrapper(path) as stub:
-                for container in stub.ListContainers(api_pb2.ListContainersRequest()).containers:
-                    if all_info:
-                        # need verbose=True to get the info which contains the PID
-                        status = stub.ContainerStatus(
-                            api_pb2.ContainerStatusRequest(container_id=container.id, verbose=True)
-                        )
-                        pid: Optional[int] = json.loads(status.info.get("info", "{}")).get("pid")
-                    else:
-                        pid = None
-
-                    containers.append(self._create_container(container, pid, rt))
+                for c in stub.ListContainers(api_pb2.ListContainersRequest()).containers:
+                    container = self._get_container_from_runtime(c.id, rt, stub, all_info)
```

Comment:
You are ignoring `all_info` here. Do not call `ContainerStatus` if you don't need

Reply:
The `get_container` API always calls `ContainerStatus` and  uses `all_info` to control the verbose filed of the `ContainerStatusRequest`.
I need the start_time to always be available when I list containers, otherwise I need some way to differntiate a tri-state in the container dataclass for that field
1. start_time=Some(time) 
2. start_time=None (container has not started yet)
3. start_time=<MISSING>  - didn't pass `all_info`, I don't know if started or not

Reply:
You want the time for whatever usage when **you** list containers, but when gProfiler lists, it doesn't require, so don't add it in this API, add it in another API or expose it with a parameter that keeps the old behavior default. You're adding N API calls...